File: PolicyPlus/CompiledStructures.vb
' Compiled data, more object-oriented and display-worthy than raw data from ADMX files
Public Class PolicyPlusCategory
    Public UniqueID As String
    Public Parent As PolicyPlusCategory
    Public Children As New List(Of PolicyPlusCategory)
    Public DisplayName As String
    Public DisplayExplanation As String
    Public Policies As New List(Of PolicyPlusPolicy)
    Public RawCategory As AdmxCategory
End Class
Public Class PolicyPlusProduct
    Public UniqueID As String
    Public Parent As PolicyPlusProduct
    Public Children As New List(Of PolicyPlusProduct)
    Public DisplayName As String
    Public RawProduct As AdmxProduct
End Class
Public Class PolicyPlusSupport
    Public UniqueID As String
    Public DisplayName As String
    Public Elements As New List(Of PolicyPlusSupportEntry)
    Public RawSupport As AdmxSupportDefinition
End Class
Public Class PolicyPlusSupportEntry
    Public Product As PolicyPlusProduct
    Public SupportDefinition As PolicyPlusSupport ' Only used if this entry actually points to another support definition
    Public RawSupportEntry As AdmxSupportEntry
End Class
Public Class PolicyPlusPolicy
    Public UniqueID As String
    Public Category As PolicyPlusCategory
    Public DisplayName As String
    Public DisplayExplanation As String
    Public SupportedOn As PolicyPlusSupport
    Public Presentation As Presentation
    Public RawPolicy As AdmxPolicy
End Class

File: PolicyPlus/CompiledStructures.cs
using System.Collections.Generic;

namespace PolicyPlus
{
    // Compiled data, more object-oriented and display-worthy than raw data from ADMX files
    public class PolicyPlusCategory
    {
        public string UniqueID;
        public PolicyPlusCategory Parent;
        public List<PolicyPlusCategory> Children = new List<PolicyPlusCategory>();
        public string DisplayName;
        public string DisplayExplanation;
        public List<PolicyPlusPolicy> Policies = new List<PolicyPlusPolicy>();
        public AdmxCategory RawCategory;
    }

    public class PolicyPlusProduct
    {
        public string UniqueID;
        public PolicyPlusProduct Parent;
        public List<PolicyPlusProduct> Children = new List<PolicyPlusProduct>();
        public string DisplayName;
        public AdmxProduct RawProduct;
    }

    public class PolicyPlusSupport
    {
        public string UniqueID;
        public string DisplayName;
        public List<PolicyPlusSupportEntry> Elements = new List<PolicyPlusSupportEntry>();
        public AdmxSupportDefinition RawSupport;
    }

    public class PolicyPlusSupportEntry
    {
        public PolicyPlusProduct Product;
        public PolicyPlusSupport SupportDefinition; // Only used if this entry actually points to another support definition
        public AdmxSupportEntry RawSupportEntry;
    }

    public class PolicyPlusPolicy
    {
        public string UniqueID;
        public PolicyPlusCategory Category;
        public string DisplayName;
        public string DisplayExplanation;
        public PolicyPlusSupport SupportedOn;
        public Presentation Presentation;
        public AdmxPolicy RawPolicy;
    }
}